drivers/spi/spi_ll_stm32.c

```diff
@@ -459,42 +459,73 @@ static void spi_stm32_cs_control(const struct device *dev, bool on)
 #endif
 }
 
-static void spi_stm32_complete(const struct device *dev, int status)
-{
-	const struct spi_stm32_config *cfg = dev->config;
-	SPI_TypeDef *spi = cfg->spi;
 #ifdef CONFIG_SPI_STM32_INTERRUPT
-	struct spi_stm32_data *data = dev->data;
-
+static void spi_stm32_disable_irqs(SPI_TypeDef *spi)
+{
 	ll_func_disable_int_tx_empty(spi);
 	ll_func_disable_int_rx_not_empty(spi);
 	ll_func_disable_int_errors(spi);
-#endif
 
+#if DT_HAS_COMPAT_STATUS_OKAY(st_stm32h7_spi)
+	LL_SPI_DisableIT_DXP(spi);
+	LL_SPI_DisableIT_TXTF(spi);
```

Comment:
I don't remember exactly why I disabled them. Maybe I did it as a "just in case" they get enabled in the future. I could change it, but I would rather avoid modify the logic at this point to avoid going through the tests I have and making everyone else to re-test.

Reply:
> I don't remember exactly why I disabled them. Maybe I did it as a "just in case" they get enabled in the future. I could change it, but I would rather avoid modify the logic at this point to avoid going through the tests I have and making everyone else to re-test.

That's not exactly a good answer. 
Despite all your efforts (which I think every one can appreciate), I don't want to rush into merging a follow up fix that could introduce other regressions (as we can see it is not easy to anticipate or prevent the regressions).

Here is my proposal:
- Revert https://github.com/zephyrproject-rtos/zephyr/pull/63173/commits/bffa0c6bddbc91d39f4b01baa34e3d0595760d50
- Push a new version of the FIFO patch together with a dts property that allows users to enable the feature.

This would be the safest way to bring this new feature (which I think is valuable to the driver) and avoid to disturb users and impact driver maturity.